persistence/state.go

```diff
@@ -354,17 +354,18 @@ func (p *PostgresContext) updateTransactionsTree() error {
 }
 
 func (p *PostgresContext) updateParamsTree() error {
-	paramsOrFlags, err := p.getParamsOrFlags(types.ParamsTableName, p.Height, false)
+	params, err := p.getParamsUpdated(p.Height)
 	if err != nil {
 		return err
 	}
 
-	for _, pf := range paramsOrFlags {
-		pfHash, err := pf.Hash()
+	for _, param := range params {
+		paramHash := crypto.SHA3Hash([]byte(param.String()))
+		paramBz, err := proto.Marshal(param)
```

Comment:
Instead of `proto.Marshal`, look into calling `codec.GetCodec.Marshal(param)` from our `codec` library

Reply:
After implementing this, locally tests are passing however the github action fails - I change the hashes to the ones the github workflow says it is - then the local tests fail saying the hashes should be the previous ones...

Looking into this now

Reply:
@Olshansk Also in `persistence/state.go` should this also be using the codec marshaller? As all the other update tree functions are using `proto.Marshal` directly

Reply:
> After implementing this, locally tests are passing however the github action fails - I change the hashes to the ones the github workflow says it is - then the local tests fail saying the hashes should be the previous ones...

Maybe the workflow was wrong? Passing now... still same hashes not changed them. My best guess was the local environment the workflow span up was dodgy?

Reply:
> @Olshansk Also in persistence/state.go should this also be using the codec marshaller? As all the other update tree functions are using proto.Marshal directly

Yes! Do you mind changing all of them? Eventually we'll block the use of `proto.` directly outside of the codec library: https://github.com/pokt-network/pocket/issues/239#issuecomment-1405417782

> Maybe the workflow was wrong 

I'm not really sure what cause it to be flaky. Is it passing everywhere now?

Reply:
@Olshansk yeah passing now locally and on GH - updated to use codec where proto was used directly doing changelogs then will merge 🫡